lib/validators/url_validator.rb

```diff
@@ -72,6 +82,26 @@ def validate_fragment(option, fragment)
         throw :invalid unless fragment.present? == option
       end
 
+      def validate_authority(option, url)
+        throw :invalid if option.is_a?(Regexp) && url.host !~ option
+        throw :invalid if option.is_a?(Array) && !option.include?(url.host)
+        if option.is_a?(Hash)
```

Comment:
We could perhaps merge the two conditions into one?
`        check_reserved_domains(url) if option.is_a?(Hash) && option[:reserved] == false
`